cache/file/file.go

```diff
@@ -0,0 +1,52 @@
+// Package file provides a disk based cache.
+package file
+
+import (
+	"github.com/pierrre/imageserver"
+	"io/ioutil"
+	"os"
+	"sync"
+)
+
+// Cache is a implementation of disk based cache system.
+type Cache struct {
+	Path string
+}
+
+// Get implements imageserver/cache.Cache.
+func (cache *Cache) Get(key string, params imageserver.Params) (*imageserver.Image, error) {
+	data := cache.getData(key)
+	if data == nil {
+		return nil, nil
+	}
+	im := new(imageserver.Image)
+	err := im.UnmarshalBinaryNoCopy(data)
+	if err != nil {
+		return nil, err
+	}
+	return im, nil
+}
+
+func (cache *Cache) getData(key string) []byte {
+	item, err := ioutil.ReadFile(cache.Path + "/" + key)
+	if err != nil {
+		return nil
+	}
+	return item
+}
+
+// Set implements imageserver/cache.Cache.
+func (cache *Cache) Set(key string, im *imageserver.Image, params imageserver.Params) error {
+	data, err := im.MarshalBinary()
+	if err != nil {
+		return err
+	}
+	return cache.setData(key, data)
+}
+
+func (cache *Cache) setData(key string, data []byte) error {
+	mutex := new(sync.Mutex)
+	mutex.Lock()
+	defer mutex.Unlock()
```

Comment:
I don't understand the purpose of these lines.
I agree, we should "lock" something, in order to prevent concurrent read/write on the same file.
But this will not work: this mutex is only usable in this method call.
It will not work with several calls.

What I would do:
- add a field `mu sync.RWMutex` to the `Cache` type
- call `RLock()` in `Get()`
- call `Lock()` in `Set()`

The performance will not be great, but it's the best I can do currently.